instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/environment_variables.py

```diff
@@ -0,0 +1,24 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Exclude the http send span from the tracing feature in Python.
+"""
+
+OTEL_PYTHON_ASGI_EXCLUDE_SEND_SPAN = "OTEL_PYTHON_ASGI_EXCLUDE_SEND_SPAN"
```

Comment:
I would prefer not introducing new environment variables and only have this supported manually. Would this be enough to cover your use case?

Reply:
Unfortunately no. I use fastapi and in that instrumentor's implementation this class isn't injectable. A further PR would be required to remedy that, and I guess also for any other frameworks where that's the case.

Reply:
Oh ugh, it's even worse. The `add_middleware` method takes a class not an instance, that would require a fair amount of refactoring. What is your worry about adding env vars if the default behavior remains the same?

Reply:
@omgitsaheadcrab 

> What is your worry about adding env vars if the default behavior remains the same?

Not a huge worry. The way you are doing it seems fairly unintrusive and we already have precedence for configuration capturing headers through env var.

> I use fastapi and in that instrumentor's implementation this class isn't injectable

Can't you simply add another configuration option [here](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/instrumentation/opentelemetry-instrumentation-fastapi/src/opentelemetry/instrumentation/fastapi/__init__.py#L275)? Is the refactoring that much?

Reply:
Ah, I'd misunderstood how add_middleware works. Happy to remove the env vars and make a small PR adding the config options to the fastapi instrumentor.